python/hail/context.py

```diff
@@ -68,6 +76,27 @@ def __init__(self, sc=None, appName="Hail", master=None, local='local[*]',
         self._jsql_context = self._jhc.sqlContext()
         self._sql_context = SQLContext(self.sc, self._jsql_context)
 
+        # do this at the end in case something errors, so we don't raise the above error without a real HC
+        Env._hc = self
+
+    @staticmethod
+    def get_running():
+        """Return the running Hail context in this Python session.
+
+        **Example**
+
+        .. doctest::
+            :options: +SKIP
+
+            >>> HailContext()  # oops! Forgot to bind to 'hc'
+            >>> hc = HailContext.get_running()  # recovery
+
+        Useful to recover a context that has been created but is unbound.
+
+        :rtype: :class:`.HailContext`
+        """
+        return env.hc
```

Comment:
`Env._hc` is a variable, `env.hc` is a getter that checks `Env._hc` for `None` and returns a nice error message if no context is created.

I want to use `Env._hc` in the constructor / stop, because we don't want the check there.  I believe that we do want the check here.